### ISSUE TITLE
Add constraints on language tags

### ISSUE THREAD

**Issue by labra** [shexSpec/shex#4]
It should be easy to specify a constraint that every value of a property must be a language tagged literal with in some specific language. 

For example

``` turtle
<SpanishTerm> { rdfs:label @"es" }
```

The syntax of language tags constraints could also be extended with regular expressions to be able to express language variations. For example:

``` turtle
<AnySpanishTerm> { rdfs:label @"es*" }
```

which would match `es`, `es-419` (Spanish spoken in latin-america), etc.

I am not sure if the syntax would need extra parenthesis as value sets and if we could remove the quotes, e.g.:

``` turtle
<SpanishTerm> { rdfs:label [@es] }
```

**Comment by ericprud** [comment]
* Jose Emilio Labra Gayo <notifications@github.com> [2016-05-26 03:50-0700]
> It should be easy to specify a constraint that every value of a property must be a language tagged literal with in some specific language. 
> 
> For example
> 
> ```turtle
> <SpanishTerm> { rdfs:label @es }
> ```
> 
> The syntax of language tags constraints could also be extended with regular expressions to be able to express language variations. For example:
> 
> ```turtle
> <AnySpanishTerm> { rdfs:label @"es*" }
> ```
> 
> which would match `es`, `es-419` (Spanish spoken in latin-america), etc.

I think this interacts with https://github.com/shexSpec/shex/issues/2 (UNIQUE keys) as it is a common use case to say we only want one label for any given langtag.


> ---
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/shexSpec/shex/issues/4

-- 
-ericP

office: +1.617.599.3509
mobile: +33.6.80.80.35.59

(eric@w3.org)
Feel free to forward this message to any list for any purpose other than
email address distribution.

There are subtle nuances encoded in font variation and clever layout
which can only be seen by printing this message on high-clay paper.

**Comment by labra** [comment]
Although a common use case interacts with UNIQUE (when we want to say that there are no more than one literal per language), there are other cases where we may want to talk about language tags. 

At this moment we can say that the values of `schema:name` must be language tagged strings by:

```
:Country { schema:name rdf:langString }
```

However, we can't say that they must be a language tagged strings in either English or French. Something like:

```
:Country { schema:name rdf:langString AND [@en @fr] }
```

**Comment by labra** [comment]
I think we could separate this issue in two parts:
1. Support to declare that literals associated with a predicate must have some language tags
1. Declare that there are no more than one combination of language tag/predicate

I will create a specific issue for the first part which I think we can include in ShEx 2.0, while the second part interacts with the `UNIQUE` feature.

**Comment by jimkont** [comment]
closed in 20170771 meeting with the following reason and votes
```
PROPOSE: close issue #4 as partially resolved by support of language tags is ShEx and track unique language tages in issue #47

Kat: +1
Andra: +1 
ericP: +1
Dimtiris: +1
Lucas: +1

RESOLVED!
```
